src/Raven.Studio/typescript/components/pages/database/indexes/errors/SelectIndexErrorsDropdown.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import "./SelectIndexDropdownToggle.scss";
+import { Badge, DropdownMenu, DropdownToggle, Label, UncontrolledDropdown } from "reactstrap";
+import { Checkbox } from "components/common/Checkbox";
+import { FlexGrow } from "components/common/FlexGrow";
+import { ColumnFilter, ColumnFiltersState, Updater } from "@tanstack/react-table";
+import { nameAndCount } from "components/pages/database/indexes/errors/types";
+
+interface SelectIndexErrorsDropdownProps {
+    dropdownType: "IndexName" | "Action";
+    indexesList: nameAndCount[];
+    filters: ColumnFiltersState;
+    setFilters: (updater: Updater<ColumnFiltersState>) => void;
+    isLoading?: boolean;
+}
+
+export function SelectIndexErrorsDropdown(props: SelectIndexErrorsDropdownProps) {
```

Comment:
I think that separating it into two components (based on dropdownType) would simplify logic